### ISSUE TITLE
type() function not working in python

### ISSUE THREAD

**Issue by LeoHink** [twibiral/obsidian-execute-code#89]
For some reason I cannot run the print out a data type. 

e.g. 

```run-python
print(type(35))
```

This simply does not return anything in obsidian although we would expect it to return `int`

**Comment by milan338** [comment]
I'm getting the same issue. Expected to see `<class 'int'>` but nothing is returned. This is because we're not sanitising before pushing to the DOM (which should be fixed in future), and so it just gets pushed as an empty DOM element because of the angled brackets:

![image](https://user-images.githubusercontent.com/56327606/193579664-3961a850-acb4-411c-8d72-5278b87b9187.png)

Though this behaviour is what lets us embed images in the output, might be a tricky fix.

**Comment by chlohal** [comment]
i had an idea that i wanted to ask you about, milan-- that we could change the inline plot function injection to put some control characters around the injected code.

For example, `addInlinePlotsToPython(...)` could inject something like

```py
sys.stdout.buffer.write("\u001d\n")
sys.stdout.buffer.write(__obsidian_execute_code_temp_pyplot_var__.getvalue())
sys.stdout.buffer.write("\u001d\n")
```

and then the `Outputter` could toggle sanitization based on that.

I'm not sure about the support for this in different terminals/systems/versions, but I think that it would be better than hiding, as we do now.

sorry :)

**Comment by chlohal** [comment]
That's a good option; my only reservation is that they claim permissiveness as a benefit in their README. Would something like that handle our use-case? It seems to be focused on security, which (while a priority of the plugin as a whole), isn't the reason for *this* problem. Testing it on  https://cure53.de/purify seems to not work.

**Comment by milan338** [comment]
You're right to fix this issue we need to escape the output not just sanitise. We could try sanitise / parse all output before pushing to the DOM, and if it's invalid html, we just replace `<` with `&lt;` and `>` with `&gt;` (assuming the intention is we want to keep all valid html in the output, otherwise we could set the flags before and after like you suggested and do the replacement on everything being pushed to the DOM except between the flags, but then users can't embed their own html).

**Comment by chlohal** [comment]
I like the idea for parsing output & escaping if it's not a valid HTML string. I think `<class 'int'>` *is* valid HTML, though, or would be interpreted as such by most lenient parsers. We might want to use an XML parser; that would preclude minor features like self-closing `<li>`s, but that's a minor aspect. 
Alternatively, if we use the flags, we could offer a new magic command to output HTML. This might be more performant, but we need to check.

**Comment by milan338** [comment]
I don't think`<class 'int'>` is valid html (https://validator.w3.org/nu/#textarea) since it isn't ever closed, but js will close it on its own when pushing it to DOM. Even then it's not valid, but it still gets treated as html. I think trying to do our own parsing is too complex, using some sort of flags is probably best.

**Comment by twibiral** [comment]
Resolved by #103 and released in [1.1.0](https://github.com/twibiral/obsidian-execute-code/releases/tag/1.0.0)
